Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -255,6 +261,77 @@ void RawDataReaderCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality che
     }
     h->SetStats(kFALSE);
     h->GetYaxis()->SetRangeUser(0, h->GetMaximum() * 1.5);
+  } else if (mo->getName() == "decodeError") {
+    auto* h = dynamic_cast<TH1D*>(mo->getObject());
+    if (checkResult != Quality::Null) {
+      msg = std::make_shared<TLatex>(0.2, 0.85, Form("Quality: %s", (checkResult.getName()).c_str()));
+      if (checkResult == Quality::Bad) {
+        msg->SetTextColor(kRed);
+      } else if (checkResult == Quality::Good) {
+        msg->SetTextColor(kGreen + 1);
+      }
+      msg->SetTextSize(0.03);
+      msg->SetNDC();
+      h->GetListOfFunctions()->Add(msg->Clone());
+    }
+    if (checkResult == Quality::Bad) {
+      float initialMessagePos = 0.8;
+      if (h->GetBinContent(1) > 0) {
+        msg = std::make_shared<TLatex>(0.2, initialMessagePos, "Failed to extract RDD");
+        msg->SetTextSize(0.03);
+        msg->SetNDC();
+        h->GetListOfFunctions()->Add(msg->Clone());
+        initialMessagePos -= 0.04;
```

Comment:
the same piece of code is repeated again and again. Please extract it in a method.